### PR TITLE
chore(example): upgrade kitchen sink example

### PR DESCRIPTION
**Pull request by ruisaraiva19**
This updates the kitchen sink example:
- Remix `blog` app to version `^1.7.0`.
- Vite `admin` app to version `^3.1.0`.
- Next.js `store` app to version `^12.3.0`

I boostraped a new Remix app via `npx create-remix@latest` and choose the `Vercel` deployment target with `TypeScript`.
After that, I manually copied the files and made sure to still have the `<CounterButton />` being rendered on the home page.

With Remix version >=1.6.4, we can now use the `watchPaths` option in `remix.config` to watch for changes on the UI package and trigger a live reload in Remix dev server.

---

Also, all apps styles have been normalized

new-style

### CODE REVIEW

File: examples/kitchen-sink/apps/blog/remix.config.js [hunk 1]
```diff
@@ -1,9 +1,15 @@
 /**
- * @type {import('@remix-run/dev/config').AppConfig}
+ * @type {import('@remix-run/dev').AppConfig}
  */
 module.exports = {
-  appDirectory: "app",
```
Comment (jaredpalmer) [inline examples/kitchen-sink/apps/blog/remix.config.js:5]: Need to specify correct outputs or change in turbo.json

**Review by jaredpalmer — CHANGES_REQUESTED** [review]

File: examples/kitchen-sink/apps/blog/remix.config.js (hunk 1 above)
Comment (ruisaraiva19) [reply examples/kitchen-sink/apps/blog/remix.config.js:5]: Updated!

**Review by jaredpalmer — DISMISSED** [review]

**Comment by jaredpalmer** [comment]
Resolve conflicts

**Comment by ruisaraiva19** [comment]
> Resolve conflicts

Fixed!

**Comment by dohomi** [comment]
@ruisaraiva19 does this PR solves the issue that any changes in `ui` are not recognised in the Remix app on any file change (dev mode)?

File: examples/kitchen-sink/patches/@remix-run+dev+1.4.3.patch [hunk 2]
```diff
@@ -0,0 +1,15 @@
+diff --git a/node_modules/@remix-run/dev/compiler.js b/node_modules/@remix-run/dev/compiler.js
+index 902b251..d5b4240 100644
+--- a/node_modules/@remix-run/dev/compiler.js
++++ b/node_modules/@remix-run/dev/compiler.js
+@@ -211,6 +211,10 @@ async function watch(config$1, {
+     toWatch.push(config$1.serverEntryPoint);
+   }
+ 
++  if(config$1.serverDependenciesToBundle) {
```
Comment (ruisaraiva19) [inline examples/kitchen-sink/patches/@remix-run+dev+1.4.3.patch:9]: Based on this [comment](https://github.com/remix-run/remix/issues/1193#issuecomment-1063143520).

File: examples/kitchen-sink/apps/blog/remix.config.js [hunk 3]
```diff
@@ -1,9 +1,16 @@
...
+  // When running locally in development mode, we use the built in remix
+  // server. This does not understand the vercel lambda module format,
+  // so we default back to the standard build output.
+  server: process.env.NODE_ENV === "development" ? undefined : "./server.js",
+  ignoredRouteFiles: [".*"],
+  // appDirectory: "app",
+  // assetsBuildDirectory: "public/build",
+  // serverBuildPath: "api/index.js",
+  // publicPath: "/build/",
+  serverDependenciesToBundle: ["ui"],
```
Comment (ruisaraiva19) [inline examples/kitchen-sink/apps/blog/remix.config.js:15]: This tells Remix to rebuild and refresh the page.
FYI: This only works based on the [patch](https://github.com/ruisaraiva19/turborepo/blob/ruisaraiva/upgrade-remix/examples/kitchen-sink/patches/@remix-run+dev+1.4.3.patch) done to the `@remix-run/dev/compiler.js` file.

**Comment by ruisaraiva19** [comment]
> @ruisaraiva19 does this PR solves the issue that any changes in `ui` are not recognised in the Remix app on any file change (dev mode) #1035?

@dohomi I just pushed a commit with a patch to solve the issue on this example!

**Comment by dohomi** [comment]
@ruisaraiva19 I see thanks for the example. I will wait until the patch is released

**Comment by narutosstudent** [comment]
Looks cool, thanks for the work here! :sparkling_heart:

**Comment by ruisaraiva19** [comment]
@dohomi I just pushed a commit updating Remix to version 1.6.5 where it has support for a `watchPaths` option for `remix.config`  to watch paths other than the `app` folder. We don't need the [patch](https://github.com/vercel/turborepo/pull/1076/commits/43adde3e38e5603c59d45b93f1857813baa39459) that I did before anymore.

You can [see here](https://github.com/ruisaraiva19/turborepo/blob/ruisaraiva/upgrade-remix/examples/kitchen-sink/apps/blog/remix.config.js#L15) the specific line of configuration.

**Comment by tknickman** [comment]
@ruisaraiva19 thanks for the work here - I can take a look at this today if this is in a good place for merge?

**Comment by ruisaraiva19** [comment]
@tknickman it should be ready to review

**Review by tknickman — CHANGES_REQUESTED** [review]
This looks great, thanks for tackling this! 

To get CI passing you will have to remove the `package-lock.json` from the root of the example - but otherwise this looks like it's ready to go.

**Comment by ruisaraiva19** [comment]
@tknickman the `package-lock.json` file is now removed.

**Review by tknickman — APPROVED** [review]
👍🏼 thanks!

**Pull request merged**